game-core/src/main/java/games/strategy/triplea/delegate/BattleDelegate.java

```diff
@@ -819,28 +814,28 @@ private void doScrambling() {
         continue;
       }
       boolean scrambledHere = false;
-      for (final var scramblers : scramblersByTerritoryPlayer.get(terrPlayer)) {
-        // verify that we didn't already scramble any of these units
-        final Iterator<Territory> territoryIter = scramblers.keySet().iterator();
-        while (territoryIter.hasNext()) {
-          final Territory t = territoryIter.next();
-          scramblers.get(t).getSecond().retainAll(t.getUnitCollection());
-          if (scramblers.get(t).getSecond().isEmpty()) {
-            territoryIter.remove();
-          }
-        }
-        if (scramblers.isEmpty()) {
-          continue;
-        }
+
+      final var scramblers = scramblersByTerritoryPlayer.get(terrPlayer);
```

Comment:
Thoughts on specifying the type instead of using var here? Its a bit hard to understand given that scramblers is a fairly complex type.

Reply:
I can if you prefer. It definitely provides value, but at the same time the type is very long:

Map<Territory, Tuple<Collection<Unit>, Collection<Unit>>>

Reply:
I ended up doing it. Done.

Reply:
Ah, I missed that this was already merged. I'll send a new PR to do that.

Reply:
It's a complete type-mess here, that's why I didn't say anything regarding that topic, I don't think the type actually helps with the general confusion.
Having some dedicated types instead of tuple would definitely help here

Reply:
Agreed.

Reply:
Yeah, this is definitely a case where a custom type would be useful but I think you get at least some value with specifying the type. It makes it so you don't have to go read a bunch of extra code to figure out what the type is.